src/codeu/chat/client/commandline/Chat.java

```diff
@@ -45,11 +49,19 @@ public Chat(Context context) {
   // is willing to take another command, the function will return true. If
   // the system wants to exit, the function will return false.
   //
-  public boolean handleCommand(String line) {
+  public boolean handleCommand(String line) throws IOException {
 
-    final Scanner tokens = new Scanner(line.trim());
-
-    final String command = tokens.hasNext() ? tokens.next() : "";
+    final List<String> args = new ArrayList<>();
+    final Tokenizer tokenizer = new Tokenizer(line);
+    try {
+      for (String token = tokenizer.next(); token != null; token = tokenizer.next()) {
+        args.add(token);
+      }
+    } catch (IOException e) {
+      System.out.println("ERROR: Nothing found in args, IOException thrown. " + e);
```

Comment:
You're reasoning is spot on. But, does the fact that next() threw an exception => "Nothing found in args,"? Perhaps something like "error parsing argstring"?